examples/scripts/interactive/compvis-viewer.js

```diff
@@ -0,0 +1,640 @@
+  diffBlue = ((diffBlue * percent) + startBlue).toString(16).split('.')[0]
+
+  // ensure 2 digits by color (necessary)
+  if (diffRed.length === 1) diffRed = '0' + diffRed
+  if (diffGreen.length === 1) diffGreen = '0' + diffGreen
+  if (diffBlue.length === 1) diffBlue = '0' + diffBlue
+
+  return '0x' + diffRed + diffGreen + diffBlue
+}
+
+function makeGradientArray () {
+  var gradientArray = []
+  for (var count = 0; count < 101; count++) {
+    var newColor = getGradientColor('FF0000', 'FFFFFF', '0000FF', (0.02 * count))
+    var numColor = parseInt(Number(newColor), 10)
+    gradientArray.push(numColor)
+  }
+  return gradientArray
+}
+
+var gradientArray = makeGradientArray()
+
+var ligandSele = '( not polymer or not ( protein or nucleic ) ) and not ( water or ACE or NH2 )'
+
+var pocketRadius = 0
+var pocketRadiusClipFactor = 1
+
+var cartoonRepr, neighborRepr, ligandRepr, contactRepr, pocketRepr, labelRepr, customRepr
+
+var heatMap, customPercent
+
+var struc
+var csv
+var neighborSele
+var sidechainAttached = false
+
+const csvResNumCol = 4
+const csvWtProbCol = 7
+const csvPrAaCol = 6
+const csvPrProbCol = 8
+
+var firstResNum
+
+function loadStructure (proteinFile, csvFile) {
+  struc = undefined
+  stage.setFocus(0)
+  stage.removeAllComponents()
+  clipNearRange.value = 0
+  clipRadiusRange.value = 100
+  pocketOpacityRange.value = 0
+  cartoonCheckbox.checked = true
+  customCheckbox.checked = false
+  hydrophobicCheckbox.checked = false
+  hydrogenBondCheckbox.checked = true
+  weakHydrogenBondCheckbox.checked = false
+  waterHydrogenBondCheckbox.checked = true
+  backboneHydrogenBondCheckbox.checked = true
+  halogenBondCheckbox.checked = true
+  metalInteractionCheckbox.checked = true
+  saltBridgeCheckbox.checked = true
+  cationPiCheckbox.checked = true
+  piStackingCheckbox.checked = true
+  return Promise.all([
+    stage.loadFile(proteinFile, { defaultRepresentation: true }),
+    NGL.autoLoad(csvFile, {
+      ext: 'csv',
+      delimiter: ',',
+      comment: '#',
+      columnNames: true
+    })
+  ]).then(function (ol) {
+    struc = ol[0]
+    csv = ol[1].data
+
+    var residueData = {}
+    console.log('l', csv.length)
+    for (var i = 0; i < csv.length; i++) {
+        const resNum = parseFloat(csv[i][csvResNumCol])
+        // console.log(i, resNum)
+        // 228 is undefined (2isk)
+        residueData[resNum] = csv[i]
+    }
+
+    firstResNum = parseInt(csv[0][csvResNumCol])
+
+    var heatMap = NGL.ColormakerRegistry.addScheme(function (params) {
+      this.domain = [ 0.5, 1 ]
+      this.scale = 'rwb'
+      this.mode = 'rgb'
+      var scale = this.getScale()
+      this.atomColor = function (atom) {
+        //console.log(atom.residueIndex)
+        // var index = parseFloat(residueData[atom.residueIndex][0])
+        // console.log(atom.residueIndex, index)
+        const csvRow = residueData[atom.resno]
+        const wtProb = parseFloat(csvRow[csvWtProbCol])
```

Comment:
I'm guessing this is the line that caused the error? Your CSV file doesn't contain data for every residue in the PDB file, so I *think* you just need to decide what the appropriate colour to return is when there's no data, e.g.

```
const csvRow = residueData[atom.resno] // Or `residueData[atom.chainname + atom.resno]
if (csvRow === undefined) {
  return 0xcccccc
}
// Then as you have it there
const wtProb = parseFloat(csvRow[csvWtProbCol]) 
...
```

Reply:
Got it working thanks! 
I am going to clean up the code, and add some helpful features and bug fixes. I should be done in a few hours.